### 6.1 Introducción

El método de mínimos cuadrados es significativo si el número ($n$) de ecuaciones de modelo paramétrico es mayor que el número ($u$) de parámetros. Esto significa que el ajuste de mínimos cuadrados por el método paramétrico es posible si la redundancia (grados de libertad) dada como ($n - u$) es mayor que cero. Matemáticamente, esto significa que las ecuaciones normales de mínimos cuadrados son singulares si la redundancia es menor que cero, asumiendo que el datum ya está definido para el ajuste como se discutió en el Capítulo 4. La singularidad de las ecuaciones normales en el ajuste de mínimos cuadrados también significa que la inversa de la matriz de los coeficientes de las ecuaciones normales ($N$) en los pasos de solución dados en la Sección 5.7 será imposible. Tenga en cuenta que cuanto mayor es la redundancia, más confiables son los resultados del ajuste.

En este capítulo, se dan algunos ejemplos de ajuste de mínimos cuadrados por el método paramétrico para ilustrar los pasos involucrados en el ajuste de mínimos cuadrados paramétricos. Continuando desde las derivaciones matemáticas dadas en el Capítulo 5, el resumen de los pasos de mínimos cuadrados para el ajuste de ecuaciones de modelo paramétrico se puede dar de la siguiente manera:

1. Formular las ecuaciones del modelo paramétrico ($\hat{\ell} = f(\hat{x})$). En esta formulación del modelo, cada observación ($\ell_i$) se traducirá en una ecuación que involucra solo algunos o todos (dependiendo del tipo de observable) de los parámetros desconocidos ($x_i$) y posibles valores constantes utilizando símbolos para observaciones y parámetros (consulte el Capítulo 5 sobre cómo formular las ecuaciones).

2. Formar la primera matriz de diseño ($A$), que es la matriz de las derivadas parciales de las ecuaciones paramétricas con respecto a los parámetros desconocidos. El tamaño de esta matriz será igual al número de ecuaciones o número de observaciones (como número de filas) por el número de parámetros (como número de columnas).

3. Elegir valores aproximados para los parámetros ($x^0$); estos valores son valores iniciales que se consideran lo suficientemente cercanos a lo que deberían ser sus valores finales. Generalmente, si esto no es el caso, los valores ajustados basados en estos valores iniciales se pueden usar nuevamente como valores aproximados en otra ronda de ajuste, de manera iterativa, hasta que los valores ajustados finales no sean significativamente diferentes de los valores aproximados (o iniciales) anteriores. En este caso, se dice que la solución de ajuste ha convergido. El ajuste de mínimos cuadrados puede tener que repetirse tantas veces como sea posible hasta que la diferencia entre la solución final y la anterior converja a una tolerancia específica. La mejor manera de obtener parámetros de coordenadas aproximadas es calcular las coordenadas de observaciones seleccionadas.

4. Sustituir los valores aproximados ($x^0$) en las ecuaciones paramétricas formuladas en el paso 1 para obtener los llamados valores simulados de las observaciones ($\ell^0 = f(x^0)$); usar estos valores simulados para obtener el vector de desajuste ($w = f(x^0) - \ell$) como la diferencia entre las observaciones simuladas ($\ell^0$) y las observaciones reales ($\ell$).

5. Formar la matriz cofactor ($Q$) de las observaciones (asumiendo el factor de varianza, $\sigma_0^2 = 1$) posicionando cuadrados de desviaciones estándar de las observaciones a lo largo de la diagonal principal de la matriz (en el orden en que se formulan las ecuaciones paramétricas en el paso 1). La inversa de esta matriz cofactor es la matriz de pesos de las observaciones ($P$).

6. Formar los siguientes productos matriciales utilizando las cantidades ya determinadas: $N = A^T PA$ y $u = A^T Pw$. Note que la matriz $A$ se evalúa utilizando los valores aproximados iniciales ($x^0$) de los parámetros y $N$ es una matriz cuadrada con su tamaño igual al número de parámetros desconocidos.

7. Resolver el sistema de ecuaciones normales para obtener las correcciones ($\delta$) a los parámetros aproximados ($x^0$) como $\delta = -N^{-1}u$.

8. Determinar los valores de los parámetros ajustados como $\hat{x} = x^0 + \delta$.

9. Si es necesario, determinar los residuales ($v$) de las observaciones y las observaciones ajustadas ($\hat{\ell}$) como $v = A\delta + w$ y $\hat{\ell} = \ell + v$, respectivamente.

### **6.2 Ejemplos Básicos de Ajuste Paramétrico**

Se dan ejemplos básicos en esta sección para ilustrar los pasos de ajuste de mínimos cuadrados paramétricos formulados en el Capítulo 5 y resumidos en la Sección 6.2. En algunos de los ejemplos, se omiten los sombreros en sus formulaciones de modelos, lo cual es aceptable por brevedad ya que sus soluciones son de mayor interés.

### **6.2.1 Ajuste de Nivelación**

Determine las elevaciones ajustadas $h_A$ y $h_B$ de los puntos A y B, respectivamente, mediante el ajuste de mínimos cuadrados paramétrico de las notas de nivelación de campo en la Tabla 6.1, que se relaciona con la red de nivelación en la Figura 6.1. Suponga que las elevaciones de los puntos FH1 y FH2 son conocidas y están fijadas en $h_1 = 100.000$ m y $h_2 = 99.729$ m, respectivamente.

**Solución:**

Las elevaciones conocidas de FH1 y FH2 definen el datum.

$$ \hat{x} = \begin{bmatrix} \hat{h}_A \\ \hat{h}_B \end{bmatrix}; \quad \hat{\ell} = \begin{bmatrix} \hat{dh}_1 \\ \hat{dh}_2 \\ \hat{dh}_3 \end{bmatrix} $$

**Tabla 6.1** Notas de campo de nivelación diferencial.

| Leq $i$ | BS | FS | Diferencia de elevación $dh_i$ (m) | Desviación estándar (m) |
|-----------|----|----|--------------------------------------|-------------------------|
| 1         | FH1| BMA| -7.341                               | 0.008                   |
| 2         | BMA| BMB| 2.495                                | 0.005                   |
| 3         | BMB| FH2| 5.107                                | 0.004                   |

**Figura 6.1** Red de nivelación diferencial de muestra.

Determine las elevaciones aproximadas de los puntos A y B usando cualquier dirección de nivelación:

$$ x^0 = \begin{bmatrix} h_A^0 = 92.659 \\ h_B^0 = 95.154 \end{bmatrix} $$

Forme el modelo paramétrico (recuerde del Capítulo 5): $\hat{\ell} = f(\hat{x})$

$$ dh_1 = \hat{h}_A - 100 \tag{6.1} $$
$$ dh_2 = \hat{h}_B - \hat{h}_A \tag{6.2} $$
$$ dh_3 = 99.729 - \hat{h}_B \tag{6.3} $$

Encuentre la matriz $A$ a partir de las Ecuaciones (6.1) a (6.3) (con los sombreros ignorados por brevedad):

$$ A = \frac{\partial \ell}{\partial x} \rightarrow A = \begin{bmatrix} 1 & 0 \\ -1 & 1 \\ 0 & -1 \end{bmatrix} $$

Como se puede ver en la matriz $A$ anterior, el número de columnas corresponde con el número de parámetros desconocidos ($h_A$ y $h_B$) en ese orden, y el número de filas corresponde con el número de ecuaciones. El tamaño de la matriz $A$, por lo tanto, es siempre el mismo que el número de ecuaciones (filas) por el número de parámetros (columnas). Forme el vector de desajuste $w$ sustituyendo los valores aproximados de los parámetros $x^0$ y las observaciones originales $\ell$ en las Ecuaciones (6.1)-(6.3):

$$ w = f(x^0) - \ell \quad \ell = \begin{bmatrix} dh_1 \\ dh_2 \\ dh_3 \end{bmatrix} = \begin{bmatrix} -7.341 \\ 2.495 \\ 5.107 \end{bmatrix} $$

La función $f(x^0)$ de arriba da lo siguiente:

$$ f(x^0) = \begin{bmatrix} -7.341 \\ 2.495 \\ 4.575 \end{bmatrix} $$

Forme la matriz de pesos ($P$) de las diferencias de elevación usando las desviaciones estándar dadas de las diferencias de elevación:

$$ P = \begin{bmatrix} 1/(0.008)^2 & 0 & 0 \\ 0 & 1/(0.005)^2 & 0 \\ 0 & 0 & 1/(0.004)^2 \end{bmatrix} $$

Forme los siguientes productos:

$$ u = A^T P w = \begin{bmatrix} 0.000 \\ -0.532 \end{bmatrix}, \quad N = A^T P A = \begin{bmatrix} 55625 & -40000 \\ -40000 & 102500 \end{bmatrix} $$

Encuentre el vector de solución:

$$ \delta = \begin{bmatrix} \delta_{h_A} \\ \delta_{h_B} \end{bmatrix} = -N^{-1}u $$

$$ \delta = \begin{bmatrix} \delta_{h_A} \\ \delta_{h_B} \end{bmatrix} = \begin{bmatrix} -0.324 \\ -0.451 \end{bmatrix} $$

Las elevaciones ajustadas de BMA y BMB se dan como sigue: $\hat{x} = x^0 + \delta$

$$ \hat{h}_A = h_A^0 + \delta_{h_A} = 92.659 - 0.324 \rightarrow 92.335 \, \text{m} $$

$$ \hat{h}_B = h_B^0 + \delta_{h_B} = 95.154 - 0.451 \rightarrow 94.703 \, \text{m} $$

Calcule los residuales de observación: $v = A\delta + w$

$$ v = A\delta + w = \begin{bmatrix} 1 & 0 \\ -1 & 1 \\ 0 & -1 \end{bmatrix} \begin{bmatrix} -0.324 \\ -0.451 \end{bmatrix} + \begin{bmatrix} 0.000 \\ 0.000 \\ -0.532 \end{bmatrix} \rightarrow v = \begin{bmatrix} -0.324 \\ -0.127 \\ -0.081 \end{bmatrix} $$

Calcule las observaciones ajustadas como sigue: $\hat{\ell} = \ell + v$

$$ \hat{dh}_1 = dh_1 + v_1 \rightarrow -7.341 - 0.324 = -7.665 \, \text{m} $$

$$ \hat{dh}_2 = dh_2 + v_2 \rightarrow 2.495 - 0.127 = 2.368 \, \text{m} $$

$$ \hat{dh}_3 = dh_3 + v_3 \rightarrow 5.107 - 0.081 = 4.494 \, \text{m} $$

Las observaciones ajustadas son:

$$ \hat{\ell} = \begin{bmatrix} \hat{dh}_1 \\ \hat{dh}_2 \\ \hat{dh}_3 \end{bmatrix} = \begin{bmatrix} -7.665 \\ 2.368 \\ 4.494 \end{bmatrix} \, \text{m} $$

In [2]:
import numpy as np
from sympy import symbols, Matrix, diff

# Definir las variables hA y hB
hA, hB = symbols('hA hB')

# Crear las ecuaciones paramétricas (6.1) a (6.3)
dh1 = hA - 100
dh2 = hB - hA
dh3 = 99.729 - hB

# Determinar la Jacobiana de las ecuaciones con respecto a hA y hB
J = Matrix([dh1, dh2, dh3]).jacobian([hA, hB])

# Valores aproximados iniciales
hA0 = 92.659
hB0 = 95.154

# Evaluar la Jacobiana como la matriz A
A = np.array(J.evalf(subs={hA: hA0, hB: hB0})).astype(np.float64)

# Observaciones dadas
L = np.array([-7.341, 2.495, 5.107])

# Determinar el vector de desajuste w
L0 = np.array([-7.341, 2.495, 4.575])
w = L0 - L

# Matriz de pesos P
P = np.diag([1/0.008**2, 1/0.005**2, 1/0.004**2])

# Calcular los productos matriciales
u = A.T @ P @ w
N = A.T @ P @ A

# Solución del vector d
d = np.linalg.inv(N) @ u

# Determinar los parámetros ajustados
hAa = hA0 + d[0]
hBa = hB0 + d[1]

# Determinar los residuales y las observaciones ajustadas
v = A @ d + w
La = L0 + v

# Imprimir resultados
print(f"d = {d}")
print(f"hAa = {hAa}")
print(f"hBa = {hBa}")
print(f"v = {v}")
print(f"La = {La}")

d = [0.32426667 0.45093333]
hAa = 92.98326666666668
hBa = 95.60493333333334
v = [ 0.32426667  0.12666667 -0.98293333]
La = [-7.01673333  2.62166667  3.59206667]


In [3]:
import numpy as np

# Definir las variables aproximadas iniciales
hA0 = 92.659
hB0 = 95.154

# Crear las ecuaciones paramétricas (6.1) a (6.3)
dh1 = hA0 - 100
dh2 = hB0 - hA0
dh3 = 99.729 - hB0

# Formar la matriz A
A = np.array([[1, 0], [-1, 1], [0, -1]])

# Observaciones dadas
L = np.array([-7.341, 2.495, 5.107])

# Determinar el vector de desajuste w
L0 = np.array([-7.341, 2.495, 4.575])
w = L0 - L

# Matriz de pesos P
P = np.diag([1/0.008**2, 1/0.005**2, 1/0.004**2])

# Calcular los productos matriciales
u = A.T @ P @ w
N = A.T @ P @ A

# Solución del vector d
d = np.linalg.inv(N) @ u

# Determinar los parámetros ajustados
hAa = hA0 + d[0]
hBa = hB0 + d[1]

# Determinar los residuales y las observaciones ajustadas
v = A @ d + w
La = L0 + v

# Imprimir resultados
print(f"d = {d}")
print(f"hAa = {hAa}")
print(f"hBa = {hBa}")
print(f"v = {v}")
print(f"La = {La}")

d = [0.32426667 0.45093333]
hAa = 92.98326666666668
hBa = 95.60493333333334
v = [ 0.32426667  0.12666667 -0.98293333]
La = [-7.01673333  2.62166667  3.59206667]


### **6.2.2 Ajuste de Estación**

El ajuste de estación es un procedimiento que consiste en encontrar los valores ajustados de ángulos medidos en una estación de levantamiento para que sean consistentes con la geometría asociada. El siguiente ejemplo se da para ilustrar la forma más simple de ajuste de estación, que puede extenderse a otras formas de ajustes de estación, como se ilustra en los ejemplos siguientes. Por ejemplo, un ángulo se midió repetidamente durante dos noches. Los valores medios aritméticos de las mediciones realizadas en las dos noches ($\bar{\ell}_1, \bar{\ell}_2$) con los segundos dados solo y los números de repeticiones ($n_1, n_2$) de las mediciones para cada noche son los siguientes:

Noche 1: $\bar{\ell}_1 = 40'' \quad n_1 = 3$  
Noche 2: $\bar{\ell}_2 = 45'' \quad n_2 = 4$

Determine el ángulo ajustado de mínimos cuadrados para las dos noches usando el método de ajuste de mínimos cuadrados paramétrico con los números de repeticiones considerados como pesos.

**Solución:**

Considere la Figura 6.3 para la ilustración del problema en este ejemplo. Los medios de las mediciones del mismo ángulo ($\theta$) se dan como $\bar{\ell}_1$ y $\bar{\ell}_2$; ahora es necesario identificar a qué referirse como parámetros ya que el problema no se formulará como ecuaciones de modelo condicional (ecuaciones que involucran solo observaciones). Dado que las direcciones se utilizan para obtener ángulos, permita que las dos direcciones involucradas para este ángulo $\theta$ sean $d_1$ (la dirección de espaldas) y $d_2$ (la dirección de frente). Dado que la lectura de la dirección de espaldas generalmente se establece en cero por conveniencia, configure $d_1$ a 0° y $d_2$ como la dirección desconocida (parámetro) a determinar mediante el método de ajuste de mínimos cuadrados paramétrico.

Note que este enfoque se puede extender para involucrar muchos ángulos medidos alrededor de un punto, como se muestra en los ejemplos siguientes.

Sea el parámetro $x = [d_2]$:

**Observaciones: $\ell = [\bar{\ell}_1, \bar{\ell}_2]$**

Formular las ecuaciones paramétricas ($\ell = f(x)$) de la siguiente manera. Dado que hay dos observaciones independientes involucradas, habrá dos ecuaciones formadas de la siguiente manera (una ecuación para cada observación con los sombreros ignorados por brevedad):

$$ \bar{\ell}_1 = d_2 - 0.0 \tag{6.4} $$
$$ \bar{\ell}_2 = d_2 - 0.0 \tag{6.5} $$

La primera matriz de diseño ($A$) se puede formar encontrando las derivadas parciales de las Ecuaciones (6.4) y (6.5) con respecto al parámetro desconocido $d_2$ como sigue:

$$ A = \frac{\partial f}{\partial x} \rightarrow A = \begin{bmatrix} 1 \\ 1 \end{bmatrix} $$

Sea $x^0$ (valor aproximado del parámetro) = 40''; sustituya este valor en las Ecuaciones (6.4) y (6.5) para obtener:

$$ f(x^0) = \begin{bmatrix} 40 - 0 \\ 40 - 0 \end{bmatrix} \rightarrow \begin{bmatrix} 40 \\ 40 \end{bmatrix} $$

Formar el vector de desajuste $w$:

$$ w = f(x^0) - \ell = \begin{bmatrix} 40 - 40 \\ 40 - 45 \end{bmatrix} \rightarrow \begin{bmatrix} 0 \\ -5 \end{bmatrix} $$

Dado que la pregunta requiere que el número de repeticiones se tome como correspondiente al peso de cada media de medición, la matriz de pesos ($P$) de las observaciones se puede dar como:

$$ P = \begin{bmatrix} 3 & 0 \\ 0 & 4 \end{bmatrix} $$

Otros productos son los siguientes:

$$ N = A^T P A \rightarrow 7 $$
$$ u = A^T P w \rightarrow -20 $$

Vector de correcciones al parámetro aproximado:

$$ \delta = -N^{-1} u \rightarrow \left( \frac{1}{7} \right) \times 20 \rightarrow 2.86'' $$

La dirección ajustada $ \hat{d}_2 $:

$$ \hat{d}_2 = x^0 + \delta $$
$$ \hat{d}_2 = 40 + 2.86 \rightarrow 42.86'' $$

El vector de residuales $v$: $v = A\delta + w$

$$ v = \begin{bmatrix} 1 \\ 1 \end{bmatrix} [2.86] + \begin{bmatrix} 0 \\ -5 \end{bmatrix} \rightarrow v = \begin{bmatrix} 2.86 \\ -2.14 \end{bmatrix} $$

El ángulo ajustado: $\hat{\ell} = \ell + v$

$$ \hat{\ell} = \begin{bmatrix} 40 \\ 45 \end{bmatrix} + \begin{bmatrix} 2.86 \\ -2.14 \end{bmatrix} \rightarrow \begin{bmatrix} 42.86 \\ 42.86 \end{bmatrix} $$

Ángulo ajustado $\theta = 42.86''$

### **Ejemplo 6.2**

Se miden tres ángulos horizontales con un instrumento de estación total alrededor del horizonte en el punto A como se muestra en la Figura 6.4. Las mediciones y sus desviaciones estándar se dan en la Tabla 6.4. Ajuste las mediciones utilizando el método de ajuste de mínimos cuadrados paramétrico.

**Tabla 6.4** Medición de ángulos horizontales alrededor del horizonte.

| Ángulo | Medición          | Desviación estándar |
|--------|-------------------|---------------------|
| $\alpha$ | $42^\circ 12' 13''$ | $3''$               |
| $\beta$  | $59^\circ 56' 15''$ | $2''$               |
| $\gamma$ | $257^\circ 51' 35''$ | $4''$               |

**Solución:**

Si se toma la dirección de la estación total A–B como valor fijo $0^\circ 00' 00''$, las otras direcciones (A–C y A–D) se consideran como los parámetros desconocidos ($x$), y los ángulos horizontales medidos son las observaciones ($\ell$). Sean los parámetros desconocidos A–C = $d_1$ y A–D = $d_2$. Las ecuaciones paramétricas ($\hat{\ell} = f(\hat{x})$) para el problema se pueden dar como sigue:

$$ \hat{\alpha} = d_1 - 0 \tag{6.6} $$
$$ \hat{\beta} = d_2 - d_1 \tag{6.7} $$
$$ \hat{\gamma} = 360^\circ - d_2 \tag{6.8} $$

La matriz A se deriva de las Ecuaciones (6.6)–(6.8) (con los sombreros ignorados por brevedad) de la siguiente manera:

$$ A = \begin{bmatrix} \frac{\partial \alpha}{\partial d_1} & \frac{\partial \alpha}{\partial d_2} \\ \frac{\partial \beta}{\partial d_1} & \frac{\partial \beta}{\partial d_2} \\ \frac{\partial \gamma}{\partial d_1} & \frac{\partial \gamma}{\partial d_2} \end{bmatrix} \rightarrow A = \begin{bmatrix} 1 & 0 \\ -1 & 1 \\ 0 & -1 \end{bmatrix} $$

Use las mediciones de ángulos dadas para determinar las direcciones iniciales (parámetros aproximados, $x^0$), y úselas en las Ecuaciones (6.6)–(6.8) para simular las observaciones; se obtienen los siguientes:

$$ x^0 = \begin{bmatrix} 42^\circ 12' 13'' \\ 102^\circ 8' 28'' \end{bmatrix}, \quad f(x^0) = \begin{bmatrix} 42^\circ 12' 13'' \\ 59^\circ 56' 15'' \\ 257^\circ 51' 32'' \end{bmatrix} $$

**Vector de desajuste:**

$$ w = f(x^0) - \ell = \begin{bmatrix} 42^\circ 12' 13'' \\ 59^\circ 56' 15'' \\ 257^\circ 51' 32'' \end{bmatrix} - \begin{bmatrix} 42^\circ 12' 13'' \\ 59^\circ 56' 15'' \\ 257^\circ 51' 35'' \end{bmatrix} \rightarrow \begin{bmatrix} 0 \\ 0 \\ -3'' \end{bmatrix} $$

**Matriz de pesos (P):**

$$ P = \begin{bmatrix} 1/9 & 0 & 0 \\ 0 & 1/4 & 0 \\ 0 & 0 & 1/16 \end{bmatrix} \quad \text{o} \quad P = \begin{bmatrix} 0.1111 & 0 & 0 \\ 0 & 0.2500 & 0 \\ 0 & 0 & 0.0625 \end{bmatrix} $$

Forme la matriz de coeficientes de las ecuaciones normales ($N$) y su inversa:

$$ N = A^T P A = \begin{bmatrix} 0.3611 & -0.2500 \\ -0.2500 & 0.3125 \end{bmatrix}, \quad N^{-1} = \begin{bmatrix} 6.2069 & 4.9655 \\ 4.9655 & 7.1724 \end{bmatrix} $$

**Parámetros ajustados por mínimos cuadrados paramétricos:**

$$ u = A^T P w = \begin{bmatrix} 0.0 \\ 0.1875 \end{bmatrix} $$

**Corrección al parámetro aproximado:**

$$ \delta = -N^{-1}u = \begin{bmatrix} -0.93 \\ -1.34 \end{bmatrix} $$

**Parámetro ajustado:**

$$ \hat{x} = x^0 + \delta $$
$$ \hat{x} = \begin{bmatrix} 42^\circ 12' 13'' \\ 102^\circ 8' 28'' \end{bmatrix} + \begin{bmatrix} -0.93 \\ -1.34 \end{bmatrix} \rightarrow \begin{bmatrix} 42^\circ 12' 12.1'' \\ 102^\circ 8' 26.7'' \end{bmatrix} $$

**Vector residual y las observaciones ajustadas:**

**Vector residual: $v = A\delta + w$**

$$ v = \begin{bmatrix} 1 & 0 \\ -1 & 1 \\ 0 & -1 \end{bmatrix} \begin{bmatrix} -0.93 \\ -1.34 \end{bmatrix} + \begin{bmatrix} 0.0 \\ 0.0 \\ -3.0 \end{bmatrix} \rightarrow \begin{bmatrix} -0.93 \\ -0.41 \\ -1.66 \end{bmatrix} $$

**Observaciones ajustadas, que son las mediciones de ángulo ajustadas: $\hat{\ell} = \ell + v$**

$$ \hat{\ell} = \begin{bmatrix} 42^\circ 12' 13'' \\ 59^\circ 56' 15'' \\ 257^\circ 51' 32'' \end{bmatrix} + \begin{bmatrix} -0.93 \\ -0.41 \\ -1.66 \end{bmatrix} \rightarrow \begin{bmatrix} 42^\circ 12' 12.1'' \\ 59^\circ 56' 14.6'' \\ 257^\circ 51' 30.3'' \end{bmatrix} $$

### **Ejemplo 6.3**

Considere la Tabla 6.5 y la Figura 6.5 en la que se realizan cuatro conjuntos de mediciones de dirección desde la estación P a las estaciones objetivo Q, R y S. Cada conjunto consiste en dos mediciones (realizadas en las posiciones de cara I y cara II del teodolito) con nivelación y recentrado y cambio de la dirección de graduación del cero del teodolito realizada entre conjuntos. Determine las mediciones de dirección ajustadas de mínimos cuadrados paramétricos para cada objetivo.

**Tabla 6.5** Mediciones de dirección desde la estación P a las estaciones Q, R y S.

| Conjunto | Estación observada | Cara I           | Cara II          |
|----------|--------------------|------------------|------------------|
| 1        | Q                  | $0^\circ 00' 00''$ | $180^\circ 00' 05''$ |
| 1        | R                  | $37^\circ 30' 27''$ | $217^\circ 30' 21''$ |
| 1        | S                  | $74^\circ 13' 42''$ | $254^\circ 13' 37''$ |
| 2        | Q                  | $0^\circ 00' 00''$ | $180^\circ 00' 05''$ |
| 2        | R                  | $36^\circ 13' 48''$ | $216^\circ 13' 42''$ |
| 2        | S                  | $72^\circ 47' 30''$ | $252^\circ 47' 24''$ |
| 3        | Q                  | $0^\circ 00' 00''$ | $180^\circ 00' 06''$ |
| 3        | R                  | $37^\circ 31' 21''$ | $217^\circ 31' 15''$ |
| 3        | S                  | $75^\circ 14' 25''$ | $255^\circ 14' 19''$ |
| 4        | Q                  | $0^\circ 00' 00''$ | $180^\circ 00' 05''$ |
| 4        | R                  | $37^\circ 30' 28''$ | $217^\circ 30' 22''$ |
| 4        | S                  | $74^\circ 13' 45''$ | $254^\circ 13' 41''$ |

**Solución:**

Sea $x_1$ el parámetro de orientación desconocido para cada conjunto. Los parámetros combinados son $x = [\delta_0, \delta_1, \delta_2, \delta_3, d_1, d_2]^T$.

El vector de observaciones es $\ell = [\ell_1, \ell_2, \ell_3, \ell_4, \ell_5, \ell_6, \ell_7, \ell_8, \ell_9, \ell_{10}, \ell_{11}, \ell_{12}]^T$.

Las ecuaciones paramétricas se formulan como sigue (una ecuación para cada conjunto, el promedio de las mediciones de cara I y cara II):

$$ \ell_1 = \delta_0 + d_0 $$
$$ \ell_2 = \delta_0 + d_1 $$
$$ \ell_3 = \delta_0 + d_2 $$
$$ \ell_4 = \delta_1 + d_0 $$
$$ \ell_5 = \delta_1 + d_1 $$
$$ \ell_6 = \delta_1 + d_2 $$
$$ \ell_7 = \delta_2 + d_0 $$
$$ \ell_8 = \delta_2 + d_1 $$
$$ \ell_9 = \delta_2 + d_2 $$
$$ \ell_{10} = \delta_3 + d_0 $$
$$ \ell_{11} = \delta_3 + d_1 $$
$$ \ell_{12} = \delta_3 + d_2 $$

Los parámetros aproximados son:

$$ x^0 = \begin{bmatrix} 0^\circ \\ 90^\circ \\ 180^\circ \\ 270^\circ \\ 37^\circ 30' 27'' \\ 74^\circ 13' 42'' \end{bmatrix} $$

La primera matriz de diseño $A$ se puede formar encontrando las derivadas parciales de las Ecuaciones (6.9)-(6.12) con respecto al vector de parámetros desconocidos $x$:

$$ A = \frac{\partial \ell}{\partial x} = \begin{bmatrix} 1 & 0 & 0 & 0 & 0 & 0 \\ 1 & 0 & 0 & 0 & 0 & 1 \\ 1 & 0 & 0 & 0 & 1 & 0 \\ 0 & 1 & 0 & 0 & 0 & 0 \\ 0 & 1 & 0 & 0 & 0 & 1 \\ 0 & 1 & 0 & 0 & 1 & 0 \\ 0 & 0 & 1 & 0 & 0 & 0 \\ 0 & 0 & 1 & 0 & 0 & 1 \\ 0 & 0 & 1 & 0 & 1 & 0 \\ 0 & 0 & 0 & 1 & 0 & 0 \\ 0 & 0 & 0 & 1 & 0 & 1 \\ 0 & 0 & 0 & 1 & 1 & 0 \end{bmatrix} $$

El producto $A^T A$ y su inversa (asumiendo pesos iguales de unidad para todas las observaciones) se pueden dar como sigue:

$$ A^T A = \begin{bmatrix} 3 & 0 & 0 & 0 & 1 & 1 \\ 0 & 3 & 0 & 0 & 1 & 1 \\ 0 & 0 & 3 & 0 & 1 & 1 \\ 0 & 0 & 0 & 3 & 1 & 1 \\ 1 & 1 & 1 & 1 & 3 & 0 \\ 1 & 1 & 1 & 1 & 0 & 3 \end{bmatrix}, \quad (A^T A)^{-1} = \begin{bmatrix} 0.5 & 0.16 & 0.16 & 0.16 & -0.25 & -0.25 \\ 0.16 & 0.5 & 0.16 & 0.16 & -0.25 & -0.25 \\ 0.16 & 0.16 & 0.5 & 0.16 & -0.25 & -0.25 \\ 0.16 & 0.16 & 0.16 & 0.5 & -0.25 & -0.25 \\ -0.25 & -0.25 & -0.25 & -0.25 & 0.25 & 0.25 \\ -0.25 & -0.25 & -0.25 & -0.25 & 0.25 & 0.25 \end{bmatrix} $$

Formar el vector de desajuste $w$ y el producto $A^T w$ como sigue:

$$ w = f(x^0) - \ell = \begin{bmatrix} -2.5'' \\ 3.0'' \\ 4.0'' \\ -3.0'' \\ 4.0'' \\ -7.5'' \\ 1.0'' \\ 4.0'' \\ -6.0'' \\ 4.0'' \\ 4.0'' \\ -1.0'' \end{bmatrix} $$

$$ A^T w = \begin{bmatrix} 4.5'' \\ -2.5'' \\ -2.5'' \\ -10.0'' \\ 9.0'' \\ -1.0'' \end{bmatrix} $$

El vector de correcciones ($\delta$) a los parámetros aproximados se resuelve como sigue:

$$ \delta = -(A^T A)^{-1} A^T w = \begin{bmatrix} 2.1'' \\ 6.9'' \\ 4.4'' \\ 6.6'' \\ -4.8'' \\ -6.0'' \end{bmatrix} $$

Los parámetros de dirección ajustados (excluyendo los parámetros de orientación) son:

$$ \hat{x} = x^0 + \delta $$

$$ d_1 = d_1^0 + \delta_1 \rightarrow d_1 = 37^\circ 30' 27'' - 4.8'' \rightarrow 37^\circ 30' 22.2'' $$

$$ d_2 = d_2^0 + \delta_2 \rightarrow d_2 = 74^\circ 13' 42'' - 6.0'' \rightarrow 74^\circ 13' 36.0'' $$

Como se puede ver en los resultados anteriores, los valores ajustados de dirección son idénticos a los obtenidos basados en el enfoque utilizado en el Ejemplo 1.3. Por lo tanto, ese enfoque de ajuste se basa en el método de mínimos cuadrados.

### **6.2.3 Ajuste de Polígono**

Un simple ajuste de polígono se usa en esta sección para ilustrar cómo ajustar una red de polígono utilizando el método de ajuste de mínimos cuadrados paramétricos. El enfoque puede extenderse a una red de polígono más compleja con los pasos dados en esta sección aplicables de la misma manera. Para ilustración, la red de polígono ajustada en 2D en la Figura 6.6 con mediciones de campo dadas en la Tabla 6.6 debe ajustarse utilizando el método de ajuste de mínimos cuadrados paramétricos. Las mediciones consisten en dos ángulos, dos rumbos y tres distancias con sus desviaciones estándar proporcionadas. Las coordenadas de los puntos de control B y E (para ser fijados en el ajuste) y las coordenadas aproximadas de los puntos desconocidos C y D que deben determinarse se dan en la Tabla 6.7. Calcule las coordenadas ajustadas de los puntos C y D.

**Tabla 6.6** Mediciones de campo.

| Símbolo | Desde | Hasta | Medición            | Desviación estándar |
|---------|-------|-------|---------------------|---------------------|
| $\ell_1$ | C     | B     | $149^\circ 59' 45''$ | $10''$              |
| $\ell_2$ | D     | C     | $240^\circ 01' 00''$ | $10''$              |
| $\ell_3$ | D     | E     | $90^\circ 00' 00''$  | $2''$               |
| $\ell_4$ | B     | E     | $59^\circ 59' 15''$  | $2''$               |
| $\ell_5$ | C     | B     | $199.880 \, m$      | $3 \, mm + 2 \, ppm$ |
| $\ell_6$ | D     | C     | $99.900 \, m$       | $0.005 \, m$        |
| $\ell_7$ | D     | E     | $177.000 \, m$      | $0.005 \, m$        |

**Tabla 6.7** Coordenadas de puntos de control y coordenadas aproximadas de puntos desconocidos.

| Estación | X (m)   | Y (m)   |
|----------|---------|---------|
| B        | 1000.000| 1000.000|
| E        | 1400.000| 1186.500|
| C        | 1173    | 1100    |
| D        | 1223    | 1186    |

Las ecuaciones de modelo paramétrico ($\hat{\ell} = f(\hat{x})$) se formulan en el orden en que se dan las observaciones en la Tabla 6.6 como sigue:

$$ \hat{\ell_1} = \text{atan}\left(\frac{\hat{Y_D} - \hat{Y_C}}{\hat{X_D} - \hat{X_C}}\right) - \text{atan}\left(\frac{Y_B - \hat{Y_C}}{X_B - \hat{X_C}}\right) \tag{6.13} $$

$$ \hat{\ell_2} = \text{atan}\left(\frac{\hat{Y_E} - \hat{Y_D}}{\hat{X_E} - \hat{X_D}}\right) + 180^\circ - \text{atan}\left(\frac{\hat{Y_C} - \hat{Y_D}}{\hat{X_C} - \hat{X_D}}\right) \tag{6.14} $$

$$ \hat{\ell_3} = \text{atan}\left(\frac{\hat{Y_E} - \hat{Y_D}}{\hat{X_E} - \hat{X_D}}\right) \tag{6.15} $$

$$ \hat{\ell_4} = \text{atan}\left(\frac{\hat{Y_C} - Y_B}{\hat{X_C} - X_B}\right) \tag{6.16} $$

$$ \hat{\ell_5} = \left[(\hat{X_C} - X_B)^2 + (\hat{Y_C} - Y_B)^2\right]^{1/2} \tag{6.17} $$

$$ \hat{\ell_6} = \left[(\hat{X_D} - \hat{X_C})^2 + (\hat{Y_D} - \hat{Y_C})^2\right]^{1/2} \tag{6.18} $$

$$ \hat{\ell_7} = \left[(\hat{X_E} - \hat{X_D})^2 + (\hat{Y_E} - \hat{Y_D})^2\right]^{1/2} \tag{6.19} $$

Sustituya los valores de las coordenadas en la Tabla 6.7 en las Ecuaciones (6.13)-(6.19) para obtener las observaciones simuladas (o derivadas) $\ell^0 = f(x^0)$ (con los valores angulares en radianes). Las observaciones derivadas ($f(x^0)$) y las observaciones reales ($\ell$) se dan respectivamente como sigue:

$$ f(x^0) = \begin{bmatrix} 2.62153553 \\ 4.18293686 \\ 1.56797148 \\ 1.0466844 \\ 199.8224 \\ 99.4786 \\ 177.0007 \end{bmatrix}, \quad \ell = \begin{bmatrix} 2.617921156 \\ 4.189081093 \\ 1.570796327 \\ 1.04697939 \\ 199.880 \\ 99.900 \\ 177.000 \end{bmatrix} $$

El vector de desajuste se da como:

$$ w = f(x^0) - \ell = \begin{bmatrix} 3.614375E-3 \\ -6.144235E-3 \\ -2.824851E-3 \\ -2.949914E-4 \\ -5.757884E-2 \\ -4.213591E-1 \\ 7.062133E-4 \end{bmatrix} $$

Los valores diagonales de la matriz de pesos $P$ de las observaciones utilizando las desviaciones estándar dadas (con las desviaciones estándar de ángulos convertidos a radianes) se dan como sigue:

$$ P = \text{diag}\left(4.254E8, 4.254E8, 1.064E10, 1.064E10, 8.651E10, 4.0E4, 4.0E4\right) $$

La primera matriz de diseño ($A$), que es la matriz Jacobiana de las Ecuaciones (6.13)–(6.19) (ignorando los sombreros por brevedad) con respecto a los parámetros desconocidos, $x = [X_C, Y_C, X_D, Y_D]^T$, se obtiene como sigue. En las derivadas parciales siguientes, los valores derivados de las distancias medidas ($\ell_5^0, \ell_6^0, \ell_7^0$) se obtienen de $f(x^0)$ dado en la Ecuación (6.20). Los valores calculados son $\ell_5^0 = 199.8224 \, m$, $\ell_6^0 = 99.4786 \, m$, $\ell_7^0 = 177.0007 \, m$:

$$ \frac{\partial \ell_1}{\partial X_C} = -\frac{(Y_D - Y_C)}{$ell_6^0)^2} + \frac{(Y_B - Y_C)}{$ell_5^0)^2} = -0.011195 $$

$$ \frac{\partial \ell_1}{\partial Y_C} = \frac{(X_D - X_C)}{$ell_6^0)^2} - \frac{(X_B - X_C)}{$ell_5^0)^2} = 0.009385 $$

$$ \frac{\partial \ell_1}{\partial X_D} = \frac{(Y_D - Y_C)}{$ell_6^0)^2} = 0.008690 $$

$$ \frac{\partial \ell_1}{\partial Y_D} = -\frac{(X_D - X_C)}{$ell_6^0)^2} = -0.005052 $$

$$ \frac{\partial \ell_2}{\partial X_C} = -\frac{(Y_E - Y_D)}{$ell_7^0)^2} = -0.008706 $$

$$ \frac{\partial \ell_2}{\partial Y_C} = \frac{(X_E - X_D)}{$ell_7^0)^2} = 0.010702 $$

$$ \frac{\partial \ell_2}{\partial X_D} = \frac{(Y_C - Y_D)}{$ell_6^0)^2} = -0.008690 $$

$$ \frac{\partial \ell_2}{\partial Y_D} = -\frac{(X_C - X_D)}{$ell_6^0)^2} = 0.005052 $$

$$ \frac{\partial \ell_3}{\partial X_C} = -\frac{(Y_E - Y_D)}{$ell_7^0)^2} = 0.000016 $$

$$ \frac{\partial \ell_3}{\partial Y_C} = \frac{(X_E - X_D)}{$ell_7^0)^2} = 0.005650 $$

$$ \frac{\partial \ell_3}{\partial X_D} = -\frac{(Y_E - Y_D)}{$ell_7^0)^2} = -0.008706 $$

$$ \frac{\partial \ell_3}{\partial Y_D} = \frac{(X_E - X_D)}{$ell_7^0)^2} = 0.010702 $$

$$ \frac{\partial \ell_4}{\partial X_C} = -\frac{(Y_C - Y_B)}{$ell_5^0)^2} = 0.002504 $$

$$ \frac{\partial \ell_4}{\partial Y_C} = \frac{(X_C - X_B)}{$ell_5^0)^2} = -0.004333 $$

$$ \frac{\partial \ell_4}{\partial X_D} = 0.0 $$

$$ \frac{\partial \ell_4}{\partial Y_D} = 0.0 $$

$$ \frac{\partial \ell_5}{\partial X_C} = \frac{(X_C - X_B)}{\ell_5^0} = 0.865769 $$

$$ \frac{\partial \ell_5}{\partial Y_C} = \frac{(Y_C - Y_B)}{\ell_5^0} = 0.500444 $$

$$ \frac{\partial \ell_5}{\partial X_D} = 0.0 $$

$$ \frac{\partial \ell_5}{\partial Y_D} = 0.0 $$

$$ \frac{\partial \ell_6}{\partial X_C} = -\frac{(X_D - X_C)}{\ell_6^0} = -0.502620 $$

$$ \frac{\partial \ell_6}{\partial Y_C} = -\frac{(Y_D - Y_C)}{\ell_6^0} = -0.864507 $$

$$ \frac{\partial \ell_6}{\partial X_D} = \frac{(X_D - X_C)}{\ell_6^0} = 0.502620 $$

$$ \frac{\partial \ell_6}{\partial Y_D} = \frac{(Y_D - Y_C)}{\ell_6^0} = 0.864507 $$

$$ \frac{\partial \ell_7}{\partial X_C} = 0.0 $$

$$ \frac{\partial \ell_7}{\partial Y_C} = 0.0 $$

$$ \frac{\partial \ell_7}{\partial X_D} = -\frac{(X_E - X_D)}{\ell_7^0} = -1.000 $$

$$ \frac{\partial \ell_7}{\partial Y_D} = -\frac{(Y_E - Y_D)}{\ell_7^0} = -0.002825 $$

La matriz $A$ se da como sigue:

$$ A = \begin{bmatrix}
-0.011195 & 0.009385 & 0.008690 & -0.005053 \\
0.008690 & -0.005053 & -0.008706 & 0.010702 \\
0.0 & 0.0 & -0.000016 & 0.005650 \\
0.002504 & -0.004333 & 0.0 & 0.0 \\
0.865769 & 0.500444 & 0.0 & 0.0 \\
-0.502620 & -0.864507 & 0.502620 & 0.864507 \\
0.0 & 0.0 & -1.000 & -0.002825
\end{bmatrix} $$

La matriz de coeficientes de las ecuaciones normales $(N$) se da como sigue:

$$ N = A^T P A = \begin{bmatrix}
2.2711 & -1.2394 & -0.8369 & 0.4625 \\
-1.2394 & 2.9956 & 0.3604 & -0.7308 \\
-0.8369 & 0.3604 & 1.1449 & -0.4179 \\
0.4625 & -0.7308 & -0.4179 & 4.2899
\end{bmatrix} \times E+5 $$

$$ N^{-1} = \begin{bmatrix}
7.528 & 2.567 & 4.726 & 0.086 \\
2.567 & 4.4526 & 0.6747 & 0.5474 \\
4.726 & 0.6747 & 12.2689 & 0.8010 \\
0.086 & 0.5474 & 0.8010 & 2.493
\end{bmatrix} \times E-6 $$

$$ A^T P w = \begin{bmatrix}
-0.43630 \\
0.53312 \\
0.28103 \\
-2.2007
\end{bmatrix} $$

El vector de correcciones a las coordenadas aproximadas se da como sigue:

$$ \delta = -N^{-1}u = \begin{bmatrix}
0.0777 \\
-0.0239 \\
0.0016 \\
0.5007
\end{bmatrix} $$

Las coordenadas ajustadas de los puntos C y D son:

$$ \hat{x} = x^0 + \delta $$

$$ \hat{x} = \begin{bmatrix}
1173 \\
1100 \\
1223 \\
1186
\end{bmatrix} + \begin{bmatrix}
0.0777 \\
-0.0239 \\
0.0016 \\
0.5007
\end{bmatrix} = \begin{bmatrix}
1173.0777 \\
1099.9761 \\
1223.0016 \\
1186.5007
\end{bmatrix} \, m $$

**Tabla 6.8** Coorden

adas ajustadas de los puntos C y D.

| Estación | X (m)       | Y (m)       |
|----------|-------------|-------------|
| C        | 1173.0777   | 1099.9761   |
| D        | 1223.0016   | 1186.5007   |

### **Ejemplo 6.4**

Para determinar las coordenadas del punto B ($x, y$) en la Figura 6.7, se realizaron las siguientes mediciones: el ángulo $\theta$ como $60^\circ 00' 00'' \pm 5''$, la distancia $S_{AB}$ como $89.50 \pm 0.01 \, m$, y la distancia $S_{CB}$ como $601.00 \pm 0.05 \, m$.

a) Formular el modelo funcional ($\hat{\ell} = f(\hat{x})$) relacionando las mediciones con las coordenadas desconocidas del punto B.

**Solución:**

**Parámetro:**

$$ x = \begin{bmatrix} x \\ y \end{bmatrix} $$

**Observaciones:**

$$ \ell = \begin{bmatrix} \theta \\ S_{AB} \\ S_{CB} \end{bmatrix} $$

Las ecuaciones del modelo paramétrico ($\hat{\ell} = f(\hat{x})$) se formulan como sigue:

$$ \hat{\theta} = \text{atan} \left( \frac{1500 - 1000}{500 - 900} \right) + 180^\circ - \text{atan} \left( \frac{\hat{x} - 1000}{\hat{y} - 900} \right) \tag{6.21} $$

$$ \hat{S}_{AB} = \sqrt{(\hat{x} - 1000)^2 + (\hat{y} - 900)^2} \tag{6.22} $$

$$ \hat{S}_{CB} = \sqrt{(\hat{x} - 1500)^2 + (\hat{y} - 500)^2} \tag{6.23} $$

b) Formar la primera matriz de diseño, $A$, asumiendo las coordenadas aproximadas ($x^0$) del punto B como $x = 1083 \, m$ y $y = 933 \, m$.

**Solución:** (Nota: La siguiente matriz $A$ se evalúa en los valores de coordenadas aproximadas del punto B)

$$ A = \frac{\partial f}{\partial x} = \begin{bmatrix} 
\frac{-(\hat{y} - 900)}{(\hat{x} - 1000)^2 + (\hat{y} - 900)^2} & \frac{\hat{x} - 1000}{(\hat{x} - 1000)^2 + (\hat{y} - 900)^2} \\
\frac{\hat{x} - 1000}{\sqrt{(\hat{x} - 1000)^2 + (\hat{y} - 900)^2}} & \frac{\hat{y} - 900}{\sqrt{(\hat{x} - 1000)^2 + (\hat{y} - 900)^2}} \\
\frac{\hat{x} - 1500}{\sqrt{(\hat{x} - 1500)^2 + (\hat{y} - 500)^2}} & \frac{\hat{y} - 500}{\sqrt{(\hat{x} - 1500)^2 + (\hat{y} - 500)^2}}
\end{bmatrix} \tag{6.24} $$

Evaluado en:

$$ A = \begin{bmatrix} 
-4.13638E-3 & 1.04036E-2 \\
0.929247 & 0.369460 \\
-0.693674 & 0.720289 
\end{bmatrix} $$

código Python usando numpy y sympy para calcular la matriz A.

In [11]:
import sympy as sp

# Definir las variables x e y
x, y = sp.symbols('x y')

# Formular las ecuaciones paramétricas (6.21) a (6.23)
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Calcular la jacobiana de las ecuaciones y evaluarla como matriz A
J = sp.Matrix([theta, SAB, SCB]).jacobian([x, y])

# Evaluar la matriz Jacobiana en los valores aproximados
x_val = 1083.0
y_val = 933.0
A = J.evalf(subs={x: x_val, y: y_val})

print(A)

Matrix([[0.00413637503133617, -0.0104036099273001], [0.929246804657356, 0.369459572936057], [-0.693673653729036, 0.720289429411685]])


In [5]:
import numpy as np

# Valores de coordenadas aproximadas
x = 1083.0
y = 933.0

# Calcular los valores de las ecuaciones
theta = np.arctan((1500 - 1000) / (500 - 900)) + np.pi + np.arctan((x - 1000) / (y - 900))
SAB = np.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = np.sqrt((x - 1500)**2 + (y - 500)**2)

# Calcular las derivadas parciales
dtheta_dx = (y - 900) / ((x - 1000)**2 + (y - 900)**2)
dtheta_dy = -(x - 1000) / ((x - 1000)**2 + (y - 900)**2)
dSAB_dx = (x - 1000) / np.sqrt((x - 1000)**2 + (y - 900)**2)
dSAB_dy = (y - 900) / np.sqrt((x - 1000)**2 + (y - 900)**2)
dSCB_dx = (x - 1500) / np.sqrt((x - 1500)**2 + (y - 500)**2)
dSCB_dy = (y - 500) / np.sqrt((x - 1500)**2 + (y - 500)**2)

# Formar la matriz A
A_np = np.array([
    [dtheta_dx, dtheta_dy],
    [dSAB_dx, dSAB_dy],
    [dSCB_dx, dSCB_dy]
])

print(A_np)

[[ 0.00413638 -0.01040361]
 [ 0.9292468   0.36945957]
 [-0.69367365  0.72028943]]


c) Utilizando las coordenadas aproximadas del punto B ($ x^0 = \begin{bmatrix} x = 1083 \\ y = 933 \end{bmatrix} $), forma el vector de los valores de las funciones de los parámetros aproximados, $ f(x^0) $.

**Solución** (sustituir los valores aproximados en las Ecuaciones (6.21)-(6.23)):

$$
\theta^0 = 128^\circ 39' 35'' - 68^\circ 19' 04'' \rightarrow 60^\circ 20' 31'' (1.05316561 \text{ rad}) \quad \text{(6.25)}
$$

$$
S_{AB}^0 = \sqrt{(x - 1000)^2 + (y - 900)^2} \rightarrow 89.320 \text{ m} \quad \text{(6.26)}
$$

$$
S_{CB}^0 = \sqrt{(x - 1500)^2 + (y - 500)^2} \rightarrow 601.147 \text{ m} \quad \text{(6.27)}
$$

$$
f(x^0) = \begin{bmatrix}
1.05316561 \text{ rad} \\
89.320 \text{ m} \\
601.147 \text{ m}
\end{bmatrix} \quad \ell = \begin{bmatrix}
1.04719755119 \text{ rad} \\
89.50 \text{ m} \\
601.10 \text{ m}
\end{bmatrix}
$$

Vector de cierre de la malla: $ w = f(x^0) - \ell $

$$
w = \begin{bmatrix}
0.0059680581 \text{ rad} \\
-0.18 \text{ m} \\
0.047 \text{ m}
\end{bmatrix} \quad \text{(6.28)}
$$

In [15]:
import sympy as sp

# Definir las variables x e y
x, y = sp.symbols('x y')

# Formular las ecuaciones paramétricas (6.21) a (6.23)
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Calcular la jacobiana de las ecuaciones y evaluarla como matriz A
J = sp.Matrix([theta, SAB, SCB]).jacobian([x, y])

# Evaluar la matriz Jacobiana en los valores aproximados
x_val = 1083.0
y_val = 933.0
A = J.evalf(subs={x: x_val, y: y_val})

L0 = sp.Matrix([theta, SAB, SCB]).subs({x: x_val, y: y_val})
L = sp.Matrix([60 * sp.pi / 180, 89.50, 601.00])

# Calcular el vector de cierre de la malla w
w = L0 - L

# Mostrar los resultados
print("L0:")
sp.pprint(L0.evalf())
print("\nL:")
sp.pprint(L.evalf())
print("\nw:")
sp.pprint(w.evalf())


L0:
⎡3.43790621778467⎤
⎢                ⎥
⎢89.3196506934504⎥
⎢                ⎥
⎣601.147236540267⎦

L:
⎡1.0471975511966⎤
⎢               ⎥
⎢     89.5      ⎥
⎢               ⎥
⎣     601.0     ⎦

w:
⎡ 2.39070866658807 ⎤
⎢                  ⎥
⎢-0.180349306549573⎥
⎢                  ⎥
⎣0.147236540267272 ⎦


In [14]:
import sympy as sp

# Definir las variables x e y
x, y = sp.symbols('x y')

# Formular las ecuaciones paramétricas (6.21) a (6.23)
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Calcular la jacobiana de las ecuaciones y evaluarla como matriz A
J = sp.Matrix([theta, SAB, SCB]).jacobian([x, y])

# Evaluar la matriz Jacobiana en los valores aproximados
x_val = 1083.0
y_val = 933.0
A = J.evalf(subs={x: x_val, y: y_val})

L0 = sp.Matrix([theta, SAB, SCB]).subs({x: x_val, y: y_val})
L = sp.Matrix([60 * sp.pi / 180, 89.50, 601.00])

# Calcular el vector de cierre de la malla w
w = L0 - L

# Calcular las derivadas parciales para la matriz A
dtheta_dx = sp.diff(theta, x).subs({x: x_val, y: y_val})
dtheta_dy = sp.diff(theta, y).subs({x: x_val, y: y_val})
dSAB_dx = sp.diff(SAB, x).subs({x: x_val, y: y_val})
dSAB_dy = sp.diff(SAB, y).subs({x: x_val, y: y_val})
dSCB_dx = sp.diff(SCB, x).subs({x: x_val, y: y_val})
dSCB_dy = sp.diff(SCB, y).subs({x: x_val, y: y_val})

A = sp.Matrix([
    [dtheta_dx, dtheta_dy],
    [dSAB_dx, dSAB_dy],
    [dSCB_dx, dSCB_dy]
])

# Mostrar los resultados
print("L0:")
sp.pprint(L0.evalf())
print("\nL:")
sp.pprint(L.evalf())
print("\nw:")
sp.pprint(w.evalf())
print("\nA:")
sp.pprint(A.evalf())


L0:
⎡3.43790621778467⎤
⎢                ⎥
⎢89.3196506934504⎥
⎢                ⎥
⎣601.147236540267⎦

L:
⎡1.0471975511966⎤
⎢               ⎥
⎢     89.5      ⎥
⎢               ⎥
⎣     601.0     ⎦

w:
⎡ 2.39070866658807 ⎤
⎢                  ⎥
⎢-0.180349306549573⎥
⎢                  ⎥
⎣0.147236540267272 ⎦

A:
⎡0.00413637503133617  -0.0104036099273001⎤
⎢                                        ⎥
⎢ 0.929246804657356    0.369459572936057 ⎥
⎢                                        ⎥
⎣-0.693673653729036    0.720289429411685 ⎦


### d) Calcular la matriz de pesos de las observaciones.

**Solución:**

$$
P = C_{\ell}^{-1} = \begin{bmatrix}
\frac{1}{5.8761076347 \times 10^{-10}} & 0 & 0 & 0 \\
0 & \frac{1}{0.0001} & 0 & 0 \\
0 & 0 & \frac{1}{0.0025} & 0 \\
0 & 0 & 0 & 1
\end{bmatrix}
$$

$$
P = C_{\ell}^{-1} = \begin{bmatrix}
1,701,806,811.84 & 0 & 0 & 0 \\
0 & 10,000 & 0 & 0 \\
0 & 0 & 400 & 0 \\
0 & 0 & 0 & 1
\end{bmatrix}
$$

In [45]:
import sympy as sp

# Definir las variables simbólicas
x, y = sp.symbols('x y')

# Definir las ecuaciones simbólicas
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Evaluar las expresiones en las coordenadas aproximadas
x_val = 1083.0
y_val = 933.0

L0 = sp.Matrix([theta, SAB, SCB]).subs({x: x_val, y: y_val})
L = sp.Matrix([60 * sp.pi / 180, 89.50, 601.00])

# Calcular el vector de cierre de la malla w
w = L0 - L

# Calcular las derivadas parciales para la matriz A
dtheta_dx = sp.diff(theta, x).subs({x: x_val, y: y_val})
dtheta_dy = sp.diff(theta, y).subs({x: x_val, y: y_val})
dSAB_dx = sp.diff(SAB, x).subs({x: x_val, y: y_val})
dSAB_dy = sp.diff(SAB, y).subs({x: x_val, y: y_val})
dSCB_dx = sp.diff(SCB, x).subs({x: x_val, y: y_val})
dSCB_dy = sp.diff(SCB, y).subs({x: x_val, y: y_val})

A = sp.Matrix([
    [dtheta_dx, dtheta_dy],
    [dSAB_dx, dSAB_dy],
    [dSCB_dx, dSCB_dy]
])

# Formar la matriz de pesos P
P_diag_values = [(1 / (5.8761076347 * 10**(-10))), 1 / (0.01)**2, 1 / (0.05)**2]
P = sp.diag(*P_diag_values)

# Mostrar los resultados
print("L0:")
sp.pprint(L0.evalf())
print("\nL:")
sp.pprint(L.evalf())
print("\nw:")
sp.pprint(w.evalf())
print("\nA:")
sp.pprint(A.evalf())
print("\nP:")
sp.pprint(P.evalf())

L0:
⎡3.43790621778467⎤
⎢                ⎥
⎢89.3196506934504⎥
⎢                ⎥
⎣601.147236540267⎦

L:
⎡1.0471975511966⎤
⎢               ⎥
⎢     89.5      ⎥
⎢               ⎥
⎣     601.0     ⎦

w:
⎡ 2.39070866658807 ⎤
⎢                  ⎥
⎢-0.180349306549573⎥
⎢                  ⎥
⎣0.147236540267272 ⎦

A:
⎡0.00413637503133617  -0.0104036099273001⎤
⎢                                        ⎥
⎢ 0.929246804657356    0.369459572936057 ⎥
⎢                                        ⎥
⎣-0.693673653729036    0.720289429411685 ⎦

P:
⎡1701806811.86766     0       0  ⎤
⎢                                ⎥
⎢       0          10000.0    0  ⎥
⎢                                ⎥
⎣       0             0     400.0⎦


### e) Calcular los valores ajustados de los parámetros desconocidos.

**Solución:**

Calcular la matriz de coeficientes de las ecuaciones normales $ N $ y su inversa:

$$
N = A^T P A = \begin{bmatrix}
37,944.774 & -70,000.928 \\
-70,000.928 & 185,767.411
\end{bmatrix}
$$

$$
N^{-1} = \begin{bmatrix}
8.645280388 \times 10^{-5} & 3.257716991 \times 10^{-5} \\
3.257716991 \times 10^{-5} & 1.765881383 \times 10^{-5}
\end{bmatrix}
$$

El otro producto:

$$
u = A^T P w = \begin{bmatrix}
-43,724.444 \\
105,041.157
\end{bmatrix}
$$

El vector de correcciones ($\delta$) a los parámetros aproximados:

$$
\delta = -N^{-1} u = \begin{bmatrix}
0.3582 \\
-0.4305
\end{bmatrix} \text{ m}
$$

Las coordenadas ajustadas del punto B:

$$
\hat{x} = x^0 + \delta = \begin{bmatrix}
\hat{x} \\
\hat{y}
\end{bmatrix} = \begin{bmatrix}
1083.000 \\
933.000
\end{bmatrix} + \begin{bmatrix}
0.3582 \\
-0.4305
\end{bmatrix} = \begin{bmatrix}
1083.358 \\
932.570
\end{bmatrix} \text{ m}
$$

---

Si necesitas más ayuda con la traducción o con otros detalles, avísame.

In [46]:
import sympy as sp

# Definir las variables simbólicas
x, y = sp.symbols('x y')

# Definir las ecuaciones simbólicas
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Evaluar las expresiones en las coordenadas aproximadas
x_val = 1083.0
y_val = 933.0

L0 = sp.Matrix([theta, SAB, SCB]).subs({x: x_val, y: y_val})
L = sp.Matrix([60 * sp.pi / 180, 89.50, 601.00])

# Calcular el vector de cierre de la malla w
w = L0 - L

# Calcular las derivadas parciales para la matriz A
dtheta_dx = sp.diff(theta, x).subs({x: x_val, y: y_val})
dtheta_dy = sp.diff(theta, y).subs({x: x_val, y: y_val})
dSAB_dx = sp.diff(SAB, x).subs({x: x_val, y: y_val})
dSAB_dy = sp.diff(SAB, y).subs({x: x_val, y: y_val})
dSCB_dx = sp.diff(SCB, x).subs({x: x_val, y: y_val})
dSCB_dy = sp.diff(SCB, y).subs({x: x_val, y: y_val})

A = sp.Matrix([
    [dtheta_dx, dtheta_dy],
    [dSAB_dx, dSAB_dy],
    [dSCB_dx, dSCB_dy]
])

# Formar la matriz de pesos P
P_diag_values = [(1 / (5.8761076347 * 10**(-10))), 1 / (0.01)**2, 1 / (0.05)**2]
P = sp.diag(*P_diag_values)

# Calcular N = A.T * P * A
N = A.T * P * A

# Calcular u = A.T * P * w
u = A.T * P * w

# Invertir N para calcular las correcciones
N_inv = N.inv()
d = -N_inv * u

# Calcular las coordenadas ajustadas
xa = x_val + d[0]
ya = y_val + d[1]

# Mostrar los resultados
print("L0:")
sp.pprint(L0.evalf())
print("\nL:")
sp.pprint(L.evalf())
print("\nw:")
sp.pprint(w.evalf())
print("\nA:")
sp.pprint(A.evalf())
print("\nP:")
sp.pprint(P.evalf())
print("\nN:")
sp.pprint(N.evalf())
print("\nN_inv:")
sp.pprint(N_inv.evalf())
print("\nu:")
sp.pprint(u.evalf())
print("\nd:")
sp.pprint(d.evalf())
print("\nxa:")
sp.pprint(xa.evalf())
print("\nya:")
sp.pprint(ya.evalf())


L0:
⎡3.43790621778467⎤
⎢                ⎥
⎢89.3196506934504⎥
⎢                ⎥
⎣601.147236540267⎦

L:
⎡1.0471975511966⎤
⎢               ⎥
⎢     89.5      ⎥
⎢               ⎥
⎣     601.0     ⎦

w:
⎡ 2.39070866658807 ⎤
⎢                  ⎥
⎢-0.180349306549573⎥
⎢                  ⎥
⎣0.147236540267272 ⎦

A:
⎡0.00413637503133617  -0.0104036099273001⎤
⎢                                        ⎥
⎢ 0.929246804657356    0.369459572936057 ⎥
⎢                                        ⎥
⎣-0.693673653729036    0.720289429411685 ⎦

P:
⎡1701806811.86766     0       0  ⎤
⎢                                ⎥
⎢       0          10000.0    0  ⎥
⎢                                ⎥
⎣       0             0     400.0⎦

N:
⎡37944.7006000141   -70000.9149753585⎤
⎢                                    ⎥
⎣-70000.9149753585  185767.760150507 ⎦

N_inv:
⎡8.6452909942049e-5   3.25771425209768e-5⎤
⎢                                        ⎥
⎣3.25771425209768e-5  1.76587680289264e-5⎦

u:
⎡16827225.5602356 ⎤
⎢                 

### f) Calcular las observaciones ajustadas.

**Solución:**

El vector residual:

$$
v = A \delta + w = \begin{bmatrix}
-5.960 \times 10^{-3} \\
0.17377 \\
-0.5585
\end{bmatrix} + \begin{bmatrix}
+5.968 \times 10^{-3} \\
-0.18 \\
+0.147
\end{bmatrix}
$$

$$
v = \begin{bmatrix}
+7.997 \times 10^{-6} \text{ rad (1.65'')} \\
-0.0062 \text{ m} \\
-0.4115 \text{ m}
\end{bmatrix}
$$

Las observaciones ajustadas:

$$
\hat{\ell} = \ell + v = \begin{bmatrix}
60^\circ 00' 00'' \\
89.500 \\
601.000
\end{bmatrix} + \begin{bmatrix}
+7.997 \times 10^{-6} \text{ rad (1.65'')} \\
-0.0062 \text{ m} \\
-0.4115 \text{ m}
\end{bmatrix} \rightarrow \begin{bmatrix}
60^\circ 00' 02'' \\
89.494 \\
600.589
\end{bmatrix}
$$

In [47]:
import sympy as sp
import numpy as np

# Definir las variables simbólicas
x, y = sp.symbols('x y')

# Definir las ecuaciones simbólicas
theta = sp.atan((1500 - 1000) / (500 - 900)) + sp.pi + sp.atan((x - 1000) / (y - 900))
SAB = sp.sqrt((x - 1000)**2 + (y - 900)**2)
SCB = sp.sqrt((x - 1500)**2 + (y - 500)**2)

# Evaluar las expresiones en las coordenadas aproximadas
x_val = 1083.0
y_val = 933.0

L0 = sp.Matrix([theta, SAB, SCB]).subs({x: x_val, y: y_val})
L = sp.Matrix([60 * sp.pi / 180, 89.50, 601.00])

# Calcular el vector de cierre de la malla w
w = L0 - L

# Calcular las derivadas parciales para la matriz A
dtheta_dx = sp.diff(theta, x).subs({x: x_val, y: y_val})
dtheta_dy = sp.diff(theta, y).subs({x: x_val, y: y_val})
dSAB_dx = sp.diff(SAB, x).subs({x: x_val, y: y_val})
dSAB_dy = sp.diff(SAB, y).subs({x: x_val, y: y_val})
dSCB_dx = sp.diff(SCB, x).subs({x: x_val, y: y_val})
dSCB_dy = sp.diff(SCB, y).subs({x: x_val, y: y_val})

A = sp.Matrix([
    [dtheta_dx, dtheta_dy],
    [dSAB_dx, dSAB_dy],
    [dSCB_dx, dSCB_dy]
])

# Formar la matriz de pesos P
P_diag_values = [(1 / (5.8761076347 * 10**(-10))), 1 / (0.01)**2, 1 / (0.05)**2]
P = sp.diag(*P_diag_values)

# Calcular N = A.T * P * A
N = A.T * P * A

# Calcular u = A.T * P * w
u = A.T * P * w

# Invertir N para calcular las correcciones
N_inv = N.inv()
d = -N_inv * u

# Calcular las coordenadas ajustadas
xa = x_val + d[0]
ya = y_val + d[1]

# Calcular las observaciones ajustadas
v = A * d + w
La = L + v

# Mostrar los resultados
print("L0:")
sp.pprint(L0.evalf())
print("\nL:")
sp.pprint(L.evalf())
print("\nw:")
sp.pprint(w.evalf())
print("\nA:")
sp.pprint(A.evalf())
print("\nP:")
sp.pprint(P.evalf())
print("\nN:")
sp.pprint(N.evalf())
print("\nN_inv:")
sp.pprint(N_inv.evalf())
print("\nu:")
sp.pprint(u.evalf())
print("\nd:")
sp.pprint(d.evalf())
print("\nxa:")
sp.pprint(xa.evalf())
print("\nya:")
sp.pprint(ya.evalf())
print("\nv:")
sp.pprint(v.evalf())
print("\nLa:")
sp.pprint(La.evalf())


L0:
⎡3.43790621778467⎤
⎢                ⎥
⎢89.3196506934504⎥
⎢                ⎥
⎣601.147236540267⎦

L:
⎡1.0471975511966⎤
⎢               ⎥
⎢     89.5      ⎥
⎢               ⎥
⎣     601.0     ⎦

w:
⎡ 2.39070866658807 ⎤
⎢                  ⎥
⎢-0.180349306549573⎥
⎢                  ⎥
⎣0.147236540267272 ⎦

A:
⎡0.00413637503133617  -0.0104036099273001⎤
⎢                                        ⎥
⎢ 0.929246804657356    0.369459572936057 ⎥
⎢                                        ⎥
⎣-0.693673653729036    0.720289429411685 ⎦

P:
⎡1701806811.86766     0       0  ⎤
⎢                                ⎥
⎢       0          10000.0    0  ⎥
⎢                                ⎥
⎣       0             0     400.0⎦

N:
⎡37944.7006000141   -70000.9149753585⎤
⎢                                    ⎥
⎣-70000.9149753585  185767.760150507 ⎦

N_inv:
⎡8.6452909942049e-5   3.25771425209768e-5⎤
⎢                                        ⎥
⎣3.25771425209768e-5  1.76587680289264e-5⎦

u:
⎡16827225.5602356 ⎤
⎢                 